lib/ansible/modules/storage/netapp/sf_volume_manager.py

```diff
@@ -0,0 +1,395 @@
+        - new account_id for the volume
+        default: None
+
+    enable512e:
+        required: false
+        type: bool
+        description:
+        - Should the volume provide 512-byte sector emulation?
+        - required when C(state=present)
+
+    qos:
+        required: false
+        type: QoS object
+        description: Initial quality of service settings for this volume.
+        default: None
+
+    attributes:
+        required: false
+        type: dict
+        description: List of Name/Value pairs in JSON object format.
+        default: None
+
+    volume_id:
+        required: false
+        description:
+        - The ID of the volume to manage or update.
+        - In order to create multiple volumes with the same name, but different volume_ids, please declare the volume_id
+          parameter with an arbitary value. However, the specified volume_id will not be assigned to the newly created
+          volume (since it's an auto-generated property).
+        default: None
+
+    size:
+        required: false
+        description:
+        - The size of the volume in (size_unit)
+        - required when C(state = present)
+
+    size_unit:
+        required: false
+        description:
+        - The unit used to interpret the size parameter
+        choices: ['bytes', 'b', 'kb', 'mb', 'gb', 'tb', 'pb', 'eb', 'zb', 'yb']
+        default: 'gb'
+
+    access:
+        required: false
+        description:
+        - Access allowed for the volume
+        choices: ['readOnly', 'readWrite', 'locked', 'replicationTarget']
+        access_type_description:
+        - readOnly: Only read operations are allowed.
+        - readWrite: Reads and writes are allowed.
+        - locked: No reads or writes are allowed.
+        - replicationTarget: Identify a volume as the target volume for a paired set of volumes. If the volume is not paired, the access status is locked.
+        - If unspecified, the access settings of the clone will be the same as the source.
+        default: None
+
+'''
+
+EXAMPLES = """
+   - name: Create Volume
+     sf_volume_manager:
+       hostname: "{{ solidfire_hostname }}"
+       username: "{{ solidfire_username }}"
+       password: "{{ solidfire_password }}"
+       state: present
+       name: AnsibleVol
+       account_id: 3
+       enable512e: False
+       size: 1
+       size_unit: gb
+
+   - name: Update Volume
+     sf_volume_manager:
+       hostname: "{{ solidfire_hostname }}"
+       username: "{{ solidfire_username }}"
+       password: "{{ solidfire_password }}"
+       state: present
+       name: AnsibleVol
+       account_id: 3
+       new_account_id: 2
+       access: readWrite
+
+   - name: Delete Volume
+     sf_volume_manager:
+       hostname: "{{ solidfire_hostname }}"
+       username: "{{ solidfire_username }}"
+       password: "{{ solidfire_password }}"
+       state: absent
+       name: AnsibleVol
+       account_id: 2
+"""
+
+RETURN = """
+
+
+"""
+
+import logging
```

Comment:
logging and debug framework already exists for modules, it also takes care of many other options that affect this, do not roll out your own